src/Api/include/Api/Orbit.h

```diff
@@ -0,0 +1,440 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_API_ORBIT_H_
+#define ORBIT_API_ORBIT_H_
+
+#include <stdint.h>
+
+// Orbit Manual Instrumentation API.
+//
+// While dynamic instrumentation is one of Orbit's core features, manual instrumentation can also be
+// extremely useful. The macros below allow you to profile sections of functions, track "async"
+// operations, and graph interesting values directly in Orbit's main capture window.
+//
+// API Summary:
+// ORBIT_SCOPE: Profile current scope.
+// ORBIT_START/ORBIT_STOP: Profile sections inside a scope.
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: Profile time spans across scopes or threads.
+// ORBIT_ASYNC_STRING: Provide custom string for an async time span.
+// ORBIT_INT: Graph int values.
+// ORBIT_INT64: Graph int64_t values.
+// ORBIT_UINT: Graph uint32_t values.
+// ORBIT_UINT64: Graph uint64_t values.
+// ORBIT_FLOAT: Graph float values.
+// ORBIT_DOUBLE: Graph double values.
+//
+// Colors:
+// Note that all of the macros above have a "_WITH_COLOR" variant that allow users to specify
+// a custom color for time slices, async strings and graph elements. A set of predefined colors can
+// be found below, see "orbit_api_color". Set custom colors with the "orbit_api_color(0xff0000ff)"
+// syntax (rgba).
+//
+// Implementation:
+// The manual instrumentation macros call empty "ORBIT_STUB" functions that Orbit dynamically
+// instruments. For manual instrumentation to appear in your Orbit capture, make sure that symbols
+// have been loaded for the manually instrumented modules.
+//
+// Performance:
+// On Linux/Stadia, our current dynamic instrumentation implementation, which relies on uprobes and
+// uretprobes, incur some non-negligible overhead (>5us per instrumented function call). Please
+// note that instrumenting too many functions will possibly cause some noticeable performance
+// degradation. Reducing overhead is our highest priority and we are actively working on a new
+// implementation that should be at least one order of magnitude faster.
+//
+// Integration:
+// To integrate the manual instrumentation API in your code base, simply include this header file.
+//
+// Please note that this feature is still considered "experimental".
+
+// To disable manual instrumentation macros, define ORBIT_API_ENABLED as 0.
+#define ORBIT_API_ENABLED 1
+
+#if ORBIT_API_ENABLED
+
+// Call once at application start, this is only needed when using the Api from pure C code.
+#define ORBIT_API_INIT() orbit_api_init()
+
+// Call once at application exit.
+#define ORBIT_API_DEINIT() orbit_api_deinit()
```

Comment:
What happens if this is not called? Isn't cleanup done anyway?

Reply:
Yeah looking at this again, it is incomplete. I actually wanted a way to let the user unload the shared object, but then it would break the program as the cached function addresses would point to freed memory. I opted to remove this altogether as it's not really needed to explicitly unload the library at the end of the program.